src/test/java/org/zalando/nakadi/controller/EventScopeTest.java

```diff
@@ -0,0 +1,119 @@
+import org.zalando.nakadi.repository.TopicRepository;
+import org.zalando.nakadi.repository.db.EventTypeCache;
+import org.zalando.nakadi.security.AuthorizedClient;
+import org.zalando.nakadi.security.Client;
+import org.zalando.nakadi.service.EventPublisher;
+import org.zalando.nakadi.utils.EventTypeTestBuilder;
+import org.zalando.nakadi.validation.EventTypeValidator;
+
+import java.util.Collections;
+import java.util.Set;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+
+public final class EventScopeTest {
+
+    private static final String EVENT_NAME = "my-event";
+    private static final String EVENT_BATCH = "[{\"payload\": \"My Event Payload\"}]";
+    private static final Set<String> SCOPE_WRITE = Collections.singleton("oauth2.scope.write");
+    private static final String URL = "/event-types/" + EVENT_NAME + "/events";
+
+    private final ObjectMapper objectMapper = new JsonConfig().jacksonObjectMapper();
+    private final EventTypeCache eventTypeCache;
+    private final StandaloneMockMvcBuilder mvcBuilder;
+
+    public EventScopeTest() throws Exception {
+        eventTypeCache = Mockito.mock(EventTypeCache.class);
+        final EventTypeMetricRegistry eventTypeMetricRegistry = Mockito.mock(EventTypeMetricRegistry.class);
+        final EventPublisher publisher = new EventPublisher(Mockito.mock(TopicRepository.class),
+                eventTypeCache, Mockito.mock(PartitionResolver.class), Mockito.mock(Enrichment.class));
+        final EventPublishingController controller = new EventPublishingController(publisher, eventTypeMetricRegistry);
+        final MappingJackson2HttpMessageConverter jackson2HttpMessageConverter = new MappingJackson2HttpMessageConverter(objectMapper);
+
+        Mockito.when(eventTypeCache.getValidator(Matchers.any())).thenReturn(new EventTypeValidator(new EventType()));
+        Mockito.when(eventTypeMetricRegistry.metricsFor(Matchers.any())).thenReturn(new EventTypeMetrics("", new MetricRegistry()));
+        mvcBuilder = MockMvcBuilders.standaloneSetup(controller).setMessageConverters(new StringHttpMessageConverter(), jackson2HttpMessageConverter);
+    }
+
+    @Test
+    public void testScopeWrite() throws Exception {
+        mvcBuilder.setCustomArgumentResolvers(new TestHandlerMethodArgumentResolver(SCOPE_WRITE));
+        Mockito.when(eventTypeCache.getEventType(EVENT_NAME)).thenReturn(createEventWith(SCOPE_WRITE));
+
+        publishEvent(EVENT_BATCH).andExpect(MockMvcResultMatchers.status().isOk());
+    }
+
+    @Test
+    public void testNoScopeWrite() throws Exception {
+        mvcBuilder.setCustomArgumentResolvers(new TestHandlerMethodArgumentResolver(Collections.emptySet()));
+        Mockito.when(eventTypeCache.getEventType(EVENT_NAME)).thenReturn(createEventWith(SCOPE_WRITE));
+
+        try {
+            publishEvent(EVENT_BATCH);
+        } catch (Exception e) {
+            Assert.assertTrue(e.getCause() instanceof IllegalScopeException);
+        }
```

Comment:
If it doesn't throw anything this will just pass. Maybe you should check for that case as well.